internal/provider/resource_vdb_test.go

```diff
@@ -58,6 +78,79 @@ func testAccCheckDctVDBConfigBasic() string {
 	`, datasource_id)
 }
 
+func testAccCheckDctVDBBookmarkConfigBasic() string {
+	var bookmark_id, vdb_id string
+
+	// init client
+	cfg := dctapi.NewConfiguration()
+	cfg.Host = os.Getenv("DCT_HOST")
+	cfg.Scheme = "https"
+	cfg.HTTPClient = &http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}}
+	cfg.AddDefaultHeader("Authorization", "apk "+os.Getenv("DCT_KEY"))
+	client := dctapi.NewAPIClient(cfg)
+
+	// create vdb
+	provisionVDBBySnapshotParameters := dctapi.NewProvisionVDBBySnapshotParameters()
+	provisionVDBBySnapshotParameters.SetAutoSelectRepository(true)
+	provisionVDBBySnapshotParameters.SetSourceDataId(os.Getenv("DATASOURCE_ID"))
+
+	vdb_req := client.VDBsApi.ProvisionVdbBySnapshot(context.Background())
+
+	vdb_res, vdb_http_res, vdb_err := vdb_req.ProvisionVDBBySnapshotParameters(*provisionVDBBySnapshotParameters).Execute()
+	if diags := apiErrorResponseHelper(vdb_res, vdb_http_res, vdb_err); diags != nil {
+		println("An error occured during vdb creation: " + vdb_err.Error())
+		return "" // return empty config to indicate config error
+	}
+	vdb_id = *vdb_res.VdbId
+
+	// poll for vdb
+	vdb_job_res, vdb_job_err := PollJobStatus(*vdb_res.Job.Id, context.Background(), client)
+
+	if vdb_job_res == Failed || vdb_job_res == Canceled || vdb_job_res == Abandoned {
+		println("An error occured during vdb job polling " + vdb_job_err)
+		return ""
+	}
+
+	// for eventual consistency
+	time.Sleep(20 * time.Second)
+
+	//create bookmark
+	bookmark := dctapi.NewBookmarkWithDefaults()
+	bookmark.SetVdbIds([]string{vdb_id})
+	bookmark.SetName(acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	bookmark_req := client.BookmarksApi.CreateBookmark(context.Background()).Bookmark(*bookmark)
+	bk_res, bk_http_res, bk_err := bookmark_req.Execute()
+
+	if diags := apiErrorResponseHelper(bk_res, bk_http_res, bk_err); diags != nil {
+		println("An error occured during bookmark creation: " + bk_err.Error())
+		return ""
+	}
+	bookmark_id = *bk_res.Bookmark.Id
+
+	// poll for bookmark
+	bk_job_res, bk_job_err := PollJobStatus(*bk_res.Job.Id, context.Background(), client)
+
+	if bk_job_res == Failed || bk_job_res == Canceled || bk_job_res == Abandoned {
+		println("An error occured during bookmark job polling: " + bk_job_err)
+		return "" // return empty config to indicate config error
+	}
+
+	// for eventual consistency
+	time.Sleep(20 * time.Second)
```

Comment:
ditto

Reply:
Same.. I have changed this static wait function to existing PollForObjectExistence() to better match the other code.